WordPress/Classes/ViewRelated/Menus/Views/MenusHeaderView.m

```diff
@@ -0,0 +1,151 @@
+    self.menusView.delegate = self;
+    
+    self.textLabel.font = [WPFontManager systemRegularFontOfSize:13];
+    self.textLabel.backgroundColor = [UIColor clearColor];
+    self.textLabel.textColor = [WPStyleGuide greyDarken20];
+    self.textLabel.text = NSLocalizedString(@"USES", @"Menus label for describing which menu the location uses in the header.");
+}
+
+- (void)setupWithMenusForBlog:(Blog *)blog
+{
+    _blog = blog;
+    
+    [self.locationsView removeAllSelectionItems];
+    [self.menusView removeAllSelectionItems];
+    
+    self.locationsView.selectionType = MenusSelectionViewTypeLocations;
+    self.menusView.selectionType = MenusSelectionViewTypeMenus;
+    
+    for (MenuLocation *location in blog.menuLocations) {
+        MenusSelectionItem *item = [MenusSelectionItem itemWithLocation:location];
+        [self.locationsView addSelectionViewItem:item];
+    }
+    for (Menu *menu in blog.menus) {
+        MenusSelectionItem *item = [MenusSelectionItem itemWithMenu:menu];
+        [self.menusView addSelectionViewItem:item];
+    }
+}
+
+- (void)addMenu:(Menu *)menu
+{
+    [self.menusView addSelectionViewItem:[MenusSelectionItem itemWithMenu:menu]];
+}
+
+- (void)removeMenu:(Menu *)menu
+{
+    MenusSelectionItem *selectionItem = [self.menusView itemWithItemObjectEqualTo:menu];
+    if (selectionItem) {
+        [self.menusView removeSelectionItem:selectionItem];
+    }
+}
+
+- (void)setSelectedLocation:(MenuLocation *)location
+{
+    MenusSelectionItem *locationItem = [self.locationsView itemWithItemObjectEqualTo:location];
+    [self.locationsView setSelectedItem:locationItem];
+}
+
+- (void)setSelectedMenu:(Menu *)menu
+{
+    MenusSelectionItem *menuItem = [self.menusView itemWithItemObjectEqualTo:menu];
+    [self.menusView setSelectedItem:menuItem];
+}
+
+- (void)refreshMenuViewsUsingMenu:(Menu *)menu
+{
+    MenusSelectionItem *item = [self.menusView itemWithItemObjectEqualTo:menu];
+    [item notifyItemObjectWasUpdated];
+}
+
+- (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
+{
+    [super traitCollectionDidChange:previousTraitCollection];
+    
+    if (self.stackView.axis == UILayoutConstraintAxisHorizontal) {
+        // Toggle the selection on a trait collection change to a horizonal axis for the stack view
+        // this ensures both selection views are expanded if one already is
+        // otherwise the design looks odd with too much negative space
+        // see userInteractionDetectedForTogglingSelectionView:expand:
+        if (self.locationsView.selectionItemsExpanded || self.menusView.selectionItemsExpanded) {
+            if (self.locationsView.selectionItemsExpanded && !self.menusView.selectionItemsExpanded) {
+                
+                [self.menusView setSelectionItemsExpanded:YES animated:NO];
+                
+            } else  if (self.menusView.selectionItemsExpanded && !self.locationsView.selectionItemsExpanded) {
```

Comment:
As mentioned above, this could be simplified down to:

`} else if (!self.locationsView.selectionItemsExpanded) {`

PS: there's a double space after the `else` keyword.